aspnetcore/migration/22-to-30.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Migrate from ASP.NET Core 2.2 to 3.0 Preview
+author: tdykstra
+description: Learn how to migrate an ASP.NET Core 2.2 project to ASP.NET Core 3.0.
+ms.author: tdykstra
+ms.custom: mvc
+ms.date: 01/04/2019
+uid: migration/22-to-30
+---
+# Migrate from ASP.NET Core 2.2 to 3.0 Preview
+
+By [Scott Addie](https://github.com/scottaddie)
+
+This article explains how to update an existing ASP.NET Core 2.2 project to ASP.NET Core 3.0.
+
+[!INCLUDE[](~/includes/net-core-prereqs-all-3.0.md)]
+
+## Update the project file
+
+* Set the [Target Framework Moniker (TFM)](/dotnet/standard/frameworks#referring-to-frameworks) to `netcoreapp3.0`:
+
+  ```xml
+  <TargetFramework>netcoreapp3.0</TargetFramework>
+  ```
+
+* Remove any `<PackageReference>` to [Microsoft.AspNetCore.All](xref:fundamentals/metapackage)` or [Microsoft.AspNetCore.App](xref:fundamentals/metapackage-app).
+
+  **Preview 1 Known Issue** - workaround for [NuGet/Home #7342](https://github.com/NuGet/Home/issues/7342) - projects that don't start with `<Project Sdk="Microsoft.NET.Sdk.Web">` will get compiler or runtime errors due to missing `Microsoft.AspNetCore.*` assemblies. This is most often the case for test projects and class libraries. The workaround is to add the following to the *.csproj* file.
```

Comment:
*ugh* ... imo, this doesn't look good and the future tense isn't necessary ... but ok.